### ISSUE TITLE
Question about locking claims in readme

### ISSUE THREAD

**Issue by 0joshuaolson1** [sean-public/fast-skiplist#8]
(sorry for opening so many issues recently)

After seeing the code, I don't understand the locking claims in the [first paragraph of this section of the readme](https://github.com/sean-public/fast-skiplist/blob/master/README.md#better-cooperative-multitasking).

These immediately lock:
- [`Set`](https://github.com/sean-public/fast-skiplist/blob/d5ed300e3bb846cca113f4d2d45ba83bb638077f/skiplist.go#L22-L24)
- [`Get`](https://github.com/sean-public/fast-skiplist/blob/d5ed300e3bb846cca113f4d2d45ba83bb638077f/skiplist.go#L53-L55)
- [`Remove`](https://github.com/sean-public/fast-skiplist/blob/d5ed300e3bb846cca113f4d2d45ba83bb638077f/skiplist.go#L80)

**Comment by ss025** [comment]
I also have similar doubt . It is saying in comments 

```
// Set inserts a value in the list with the specified key, ordered by the key.
// If the key exists, it updates the value in the existing node.
// Returns a pointer to the new element.
// Locking is optimistic and happens only after searching.
```
But First line of function is 
```
func (list *SkipList) Set(key float64, value interface{}) *Element {
	list.mutex.Lock()
	defer list.mutex.Unlock()

        // ----
}
```
which looks like lock during entire function ,

Correct me if anything is misunderstood

**Comment by sean-public** [comment]
Closed by 1de4bd3, changes to locking were made based on multithreaded test results (i.e. f399345).
